aggregator_core/src/datastore.rs

```diff
@@ -4583,34 +4584,38 @@ WHERE task_id = $1
     //    aggregations in the batch which are in a non-failure state (START/WAITING/FINISHED).
     async fn read_batch_size(
         &self,
-        task_id: &TaskId,
+        task_pkey: i64,
         batch_id: &BatchId,
     ) -> Result<RangeInclusive<usize>, Error> {
-        // TODO(#1467): fix this to work in presence of GC.
         let stmt = self
             .prepare_cached(
                 "-- read_batch_size()
-WITH batch_report_aggregation_statuses AS
-    (SELECT report_aggregations.state, COUNT(*) AS count FROM report_aggregations
-     JOIN aggregation_jobs
+WITH report_aggregations_count AS (
+    SELECT COUNT(*) as count FROM report_aggregations
+    JOIN aggregation_jobs
         ON report_aggregations.aggregation_job_id = aggregation_jobs.id
-     WHERE aggregation_jobs.task_id = (SELECT id FROM tasks WHERE task_id = $1)
-     AND report_aggregations.task_id = aggregation_jobs.task_id
-     AND aggregation_jobs.batch_id = $2
-     GROUP BY report_aggregations.state)
+    WHERE aggregation_jobs.task_id = $1
+    AND report_aggregations.task_id = aggregation_jobs.task_id
+    AND aggregation_jobs.batch_id = $2
+    AND report_aggregations.state in ('START', 'WAITING')
+),
+batch_aggregation_count AS (
+    SELECT SUM(report_count) as count FROM batch_aggregations
```

Comment:
```suggestion
    SELECT SUM(report_count) AS count FROM batch_aggregations
```

super-nit: same as above